packages/video_player/video_player/README.md

```diff
@@ -125,6 +126,34 @@ documentation in order to give a more elaborate overview of the API.
 
 This is not complete as of now. You can contribute to this section by [opening a pull request](https://github.com/flutter/plugins/pulls).
 
+### Data Sources
+
+You can provide a video from different data sources using different constructors.
+
+- `VideoPlayerController.asset` loads a video from your Flutter assets,
+- `VideoPlayerController.file` loads a video from a local file,
+- `VideoPlayerController.network` loads a video from the network, and
+- (**Android Only**) `VideoPlayerController.contentUri` loads a video from a provided content-uri.
+
+You cannot pass a YouTube video URI into the `VideoPlayerController.network` constructor.
+Functionality to decode YouTube URIs is not provided by this plugin.
+
+To learn more about the limitations of `VideoPlayerController` constructors, see the [`Constructors` documentation](https://pub.dev/documentation/video_player/latest/video_player/VideoPlayerController-class.html#constructors).
```

Comment:
What limitations is this referring to? I don't see any particularly interesting notes in that documentation that some would need to go read in this context.